utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -171,6 +177,46 @@ def cmake_options(self, args):
         options += '-DCMAKE_ANDROID_NDK:PATH=%s' % (args.android_ndk)
         return options
 
+    def ndk_toolchain_path(self, args):
+        return '%s/toolchains/llvm/prebuilt/%s' % (
+            args.android_ndk, StdlibDeploymentTarget.host_target().name)
+
+    def swiftpm_config(self, args, output_dir, swift_toolchain, resource_path):
+        config_file = '%s/swiftpm-android-%s.json' % (output_dir, args.android_arch)
+
+        if os.path.exists(config_file):
+            return config_file
+
+        spm_json = '{\n'
+        spm_json += '  "version": 1,\n'
+        spm_json += '  "target": "%s-unknown-linux-android",\n' % args.android_arch
+        spm_json += '  "toolchain-bin-dir": "%s/bin",\n' % swift_toolchain
+        spm_json += '  "sdk": "%s/sysroot",\n' % self.ndk_toolchain_path(args)
+
+        spm_json += '  "extra-cc-flags": [ "-fPIC", "-I%s/usr/include" ],\n' % (
+                    args.cross_compile_deps_path)
```

Comment:
This include and the two flags below that use `cross_compile_deps_path` replace [only two flags that I had to pass in as part of an SPM command-line](https://github.com/buttaface/termux-packages/blob/swift55/packages/swift/swiftpm-driver-termux-flags#L18). I tracked this down to [SPM passing the command-line `cCompilerFlags` to the Swift compiler too](https://github.com/apple/swift-package-manager/blob/main/Sources/Build/BuildPlan.swift#L48), [which was added by Ankit a long time ago](https://github.com/apple/swift-package-manager/pull/889/files#diff-13eb4b18e8a4f2de8467dae905be6bdc133208159aded4aa6dbb6a03be21c6bdR39), whereas the toolchain/workspace `extraCCFlags` that I'm setting here are [only passed to the clang/clang++ compiler](https://github.com/apple/swift-package-manager/blob/main/Sources/Build/BuildPlan.swift#L333), so I had to add it again below.

@abertelrud, is this intentional? If not, we should probably make the two ways to pass in C flags consistent, though it is a gray area and I could think of contradictory reasons of precision, maintaining past behavior, and convenience to go either way. Let me know what you think of this inconsistency.

Reply:
Just saw this comment now — apologies for the delay (I've been away for a while).

I don't know whether it's intentional to be inconsistent, but I am pretty sure that the reason for passing Clang flags to the Swift compiler is that the latter needs to be able to build Clang modules on-demand, and a user-provided set of Clang flags could affect how it's built (`-I` or `-D` flags for example).

Intuitively I would have expected that all custom Clang flags, wherever they came from, would have been passed to the Swift compiler.

(I'm specifically saying "Clang flags" and not "C flags" since it really applies to all the Clang-supported languages)

Reply:
We decided to keep it the way it is in apple/swift-package-manager#3636 and I added some doc there documenting this difference.